lib/rules/jest/no-if.js

```diff
@@ -0,0 +1,93 @@
+const {docsUrl} = require('../../utilities');
+
+const TEST_FUNCTION_NAMES = [
+  'it',
+  'xit',
+  'fit',
+  'test',
+  'xtest',
+  'describe',
+  'fdescribe',
+  'xdescribe',
+];
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Prevent if statements in tests.',
+      category: 'Best Practices',
+      recommended: false,
+      uri: docsUrl('jest/no-if'),
+    },
+    messages: {
+      noIf: [
+        'Tests should not contain if statements.',
+        'This is usually an indication that you',
+        'are attempting to test too much at once',
+        'and may want to consider breaking the if',
+        'statement out into a separate test.',
+      ].join(' '),
+    },
+  },
+
+  create(context) {
+    let inCallExpression = false;
+
+    return {
+      CallExpression(node) {
+        if (notTestFunction(node) || hasEmptyBody(node)) {
+          return;
+        }
+        inCallExpression = true;
```

Comment:
The iffy thing with doing this in scope is that nesting can throw things off. Nesting might not be too much of an issue here but you still might want to go with the more formal mechanism for this (state)

Reply:
Can you clarify what you mean when you say `more formal mechanism for this (state)`?

Fixing the nesting part should be easy if I just keep track of the depth instead of just a boolean for being in a call expression of not.

Reply:
I meant having it on the `context.state` or whatever that field is called so that it is scoped to the part of the tree where you are actually inside the test block (structurally, I don't think it makes much of a difference, but that's what the feature is there for)

Reply:
Wait actually maybe I am getting mixed up with Babel plugins :P

Reply:
lol you are. I only know because I did the same thing.